lib/token.js

```diff
@@ -52,7 +52,11 @@ function decodeIdToken(idToken) {
       signature: jwt[2]
     };
   } catch(e) {
-    throw new AuthSdkError('Malformed idToken');
+    if (isIdToken) {
+      throw new AuthSdkError('Malformed idToken');
```

Comment:
This isn't something a user can do anything about, so the check should be on ANY error (maybe switching on `AuthSdkError` or `OAuthError`, but you're right that there shouldn't be a check on the string. I'll remove this logic here and change our tests to check for an error with a type of `AuthSdkError` and a non-empty errorSummary.